### DISCUSSION TITLE
Issue with page navigation in 20.0.3

### DISCUSSION THREAD

**Discussion by 400ochkov** [hardkoded/puppeteer-sharp#2813] [Q&A]
Hello, I have updated PuppeteerSharp to 20.0.3 recently and faced issue with navigation to url. I use this code:
```
var navigationTask = page.WaitForNavigationAsync(new NavigationOptions
{
    Referer = referer,
    WaitUntil = new[] { WaitUntilNavigation.Networkidle0 },
    Timeout = 40000 
});
await page.GoToAsync(link);
try
{
    await navigationTask;
}
catch
{
}
```

and sometimes nothing happen, not even single try to load page, just default chrome page. If reconnect to browser and try again it can make navigation. But sometimes everything work properly without reconnection. I have not any idea what is wrong...

**Comment by kblok** [comment]
`Networkidle0` can be flaky sometimes.

**Reply by 400ochkov** [reply]
Is Networkidle2 work properly?
